src/dmd/aggregate.d

```diff
@@ -814,3 +811,36 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
         v.visit(this);
     }
 }
+
+/*********************************
+ * Iterate this dsymbol or members of this scoped dsymbol, then
+ * call `fp` with the found symbol and `params`.
+ * Params:
+ *  symbol = the dsymbol or parent of members to call fp on
+ *  fp = function pointer to process the iterated symbol.
+ *       If it returns nonzero, the iteration will be aborted.
+ *  ctx = context parameter passed to fp.
+ * Returns:
+ *  nonzero if the iteration is aborted by the return value of fp,
+ *  or 0 if it's completed.
+ */
+int apply(Dsymbol symbol, int function(Dsymbol, void*) fp, void* ctx)
```

Comment:
Why not use the variadic parameter? It looks nicer than having `void*` in the signature. Are you worried about the size of the binary? It seems like there are 3 different instantiations only.

Reply:
@thewilsonator  ?

Reply:
Still, I think it's better to use the variadic. But I am willing to let this slide.